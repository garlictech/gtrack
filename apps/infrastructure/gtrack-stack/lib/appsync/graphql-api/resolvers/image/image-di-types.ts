export const DI_TYPES = {
  createImage: Symbol('createImage'),
  createMultipleImages: Symbol('createMultipleImages'),
  ImageModel: Symbol('ImageModel'),
  ImageValidator: Symbol('ImageValidator'),
  createImageValidator: Symbol('createImageValidator'),
  createMultipleImageValidator: Symbol('createMultipleImageValidator'),
  updateImageValidator: Symbol('updateImageValidator'),
  updateImage: Symbol('updateImage'),
  hashGenerator: Symbol('imageHashGenerator'),
};
