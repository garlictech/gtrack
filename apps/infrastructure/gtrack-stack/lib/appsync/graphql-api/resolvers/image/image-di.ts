import {
  createImageInputSchema,
  multipleObjectValidator,
  validateCreateImageInput,
  validateImage,
} from '@bit/garlictech.universal.gtrack.graphql-api';
import { ImageModel } from '@bit/garlictech.universal.gtrack.graphql-api/dynogels-models/image-model';
import 'reflect-metadata';
import { basicCrudBinder } from '../resolver-lib';
import { DI_TYPES } from './image-di-types';
import { validateUpdateImageInput } from './update-image-schema';
import { ImageFp } from '@bit/garlictech.universal.gtrack.image';

export const diBinder = (container: Container): void => {
  basicCrudBinder<CreateImageInput, UpdateImageInput, Image>({
    createSymbol: DI_TYPES.createImage,
    createMultipleSymbol: DI_TYPES.createMultipleImages,
    modelSymbol: DI_TYPES.ImageModel,
    objectValidatorSymbol: DI_TYPES.ImageValidator,
    createObjectValidatorSymbol: DI_TYPES.createImageValidator,
    createMultipleObjectValidatorSymbol: DI_TYPES.createMultipleImageValidator,
    objectModel: ImageModel,
    createObjectValidator: validateCreateImageInput,
    createMultipleObjectValidator: multipleObjectValidator<CreateImageInput>(
      createImageInputSchema
    ),
    objectValidator: validateImage,
    updateSymbol: DI_TYPES.updateImage,
    updateObjectValidatorSymbol: DI_TYPES.updateImageValidator,
    updateObjectValidator: validateUpdateImageInput,
    hashGeneratorSymbol: DI_TYPES.hashGenerator,
    hashGenerator: ImageFp.generateId,
  })(container);
};
