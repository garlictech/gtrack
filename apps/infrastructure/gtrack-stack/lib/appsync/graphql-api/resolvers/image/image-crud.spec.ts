import { Container } from 'inversify';
import * as Joi from 'joi';
import * as fp from 'lodash/fp';
import { diBinder } from './image-di';
import { DI_TYPES } from './image-di-types';

test('hash generator must pick the original url', () => {
  const container = new Container();
  diBinder(container);
  const hashGenerator: any = container.get(DI_TYPES.hashGenerator);
  expect(
    fp.pick(
      ['error', 'value'],
      Joi.validate(
        hashGenerator({
          sourceObject: { objectId: 'OBJECTID', objectType: 'OBJECTTYPE' },
        }),
        Joi.string()
      )
    )
  ).toMatchSnapshot();
});
