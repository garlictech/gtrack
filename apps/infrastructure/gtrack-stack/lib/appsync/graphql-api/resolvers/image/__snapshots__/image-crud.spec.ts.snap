// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash generator must pick the original url 1`] = `
Object {
  "error": null,
  "value": "OBJECTTYPE-OBJECTID",
}
`;
